### PR TITLE
Add several Numeric#step specs and pass them

### PR DESCRIPTION
**Pull request by havenwood**

### CODE REVIEW

File: src/main/ruby/core/numeric_mirror.rb [hunk 1]
```diff
@@ -69,6 +69,13 @@ def step_size(limit, step, by)
         end
       end
 
+      def stepping_forever?(limit, step, asc)
+        limit.nil? || step.zero? ||
+          asc  && limit == Float::INFINITY  && step != Float::INFINITY ||
+          !asc && limit == -Float::INFINITY && step != -Float::INFINITY
```
Comment (eregon) [inline src/main/ruby/core/numeric_mirror.rb]: I think a `if` would be nicer than a very long conditional here:
```ruby
return true if limit.nil? || step.zero?
if asc
  limit == Float::INFINITY  && step != Float::INFINITY
else
  limit == -Float::INFINITY && step != -Float::INFINITY
end
```

**Comment by eregon** [comment]
Thank you for the fixes and the new specs!
BTW, to run specs with MRI (to verify they pass there too), you can do:
```bash
$ jt test some/spec.rb -t ruby
```
I'll add this to https://github.com/graalvm/truffleruby/blob/master/doc/contributor/workflow.md.

File: src/main/ruby/core/numeric_mirror.rb (hunk 1 above)
Comment (havenwood) [inline src/main/ruby/core/numeric_mirror.rb]: Thanks! I refactored the method to be just as you suggest.

**Comment by chrisseaton** [comment]
The SVM failures are unrelated to your code - we're fixing it.

**Pull request merged**